serial/serial_resources/process_entry.py

```diff
@@ -0,0 +1,149 @@
+        dataset = serial.Datasets.get(dataset_name, "PARAMETRIC_QUANTITATIVE", self.process_id)
+        data = {"type": "NUMERICAL", "dataset_id": dataset.dataset_id, "value": value}
+        if usl:
+            data["usl"] = usl
+        if lsl:
+            data["lsl"] = lsl
+        return self.client.make_api_request(f"/processes/entries/{self.id}", "PUT", data=data)
+
+    def add_file(self, dataset_name, path, bucket_type, file_name=None):
+        """
+        Add file data to a process entry
+
+        Args: 
+        - dataset_name: User facing name of the dataset
+        - path: Path to the file on your file system
+        """
+        if serial.debug:
+            pass
+        if not file_name:
+            file_name = os.path.basename(path)
+        # if in the future we need to do a better guess, we can use python-magic to determine the mimetype
+        files = {'file': (file_name, open(path, 'rb'), mimetypes.guess_type(path))} 
+        storage_object = self.client.make_api_request("/files", "POST", files=files)
+        dataset_type = "FILE"
+        if bucket_type == "images":
+            dataset_type = "IMAGE"
+        dataset = serial.Datasets.get(dataset_name, dataset_type, self.process_id)
+        data = {"type": "FILE", "dataset_id": dataset.dataset_id, "bucket_type": bucket_type, "file_id": storage_object["name"], "file_name": file_name}
+        return self.client.make_api_request(f"/processes/entries/{self.id}", "PUT", data=data)
+    
+    def add_boolean(self, dataset_name, value, expected_value):
+        """
+        Add boolean data to a process entry
+
+        Args: 
+        - dataset_name: User facing name of the dataset
+        - value: The value to be submitted
+        - expected_value: argument to pass an expected value. If the 
+        values are the same, it would be considered a passing test result
+        """
+        dataset = serial.Datasets.get(dataset_name, "CHECKBOX", self.process_id)
+        data = {"type": "BOOLEAN", "dataset_id": dataset.dataset_id, "value": value, "expected_value": expected_value}
+        return self.client.make_api_request(f"/processes/entries/{self.id}", "PUT", data=data)
+    
+    def submit(self, cycle_time=None, is_pass=None, is_complete=None):
+        """
+        Mark a process entry as completed
+
+        Args: 
+        - cycle_time: Optional number for seconds elapsed since the last cycle
+        finished for this process.
+        - is_pass: Optional boolean for indicating whether the process passed for
+        this entry
+        """
+        data = {}
+        if cycle_time:
+            data["cycle_time"] = cycle_time
+        if is_pass: 
+            data["is_pass"] = is_pass
+        if is_complete:
+            data["is_complete"] = is_complete
+
+        return self.client.make_api_request(f"/processes/entries/{self.id}", "PATCH", data=data)
+
```

Comment:
In an ideal world, these would return the process Entry object with the linked data?